### ISSUE TITLE
Not detecting some packages as third-party

### ISSUE THREAD

**Issue by sambrightman** [timothycrosley/isort#882]
I depend on the distribution `pymongo`, which installs packages `pymongo` and `bson` (and `gridfs` but I don't use that). I have:

```
from __future__ import absolute_import, division, print_function

import datetime
import logging
import operator
import os
import socket

import bson
import pymongo
```

I run `python setup.py test` and get:

```
========================================================================================================================================================= FAILURES ==========================================================================================================================================================
________________________________________________________________________________________________________________________________________________ [pylint] src/deploy/ddb.py _________________________________________________________________________________________________________________________________________________
C: 10, 0: third party import "import pymongo" should be placed before "import bson" (wrong-import-order)
```

This does not occur on 4.3.10 but does on 4.3.11. I also checked that it does not occur before 4.3.5 (because I read that too many files were being ignored from that version until 4.3.11).

**Comment by sblask** [comment]
Same problem here when linting with pylint:
```
************* Module cli.partial_environment_command
cli/partial_environment_command.py:18 [wrong-import-order] third party import "import requests" should be placed before "import botocore.exceptions"
cli/partial_environment_command.py:19 [wrong-import-order] third party import "import tabulate" should be placed before "import botocore.exceptions"
cli/partial_environment_command.py:20 [wrong-import-order] third party import "import troposphere" should be placed before "import botocore.exceptions"
************* Module cli.overview_command
cli/overview_command.py:11 [wrong-import-order] third party import "import tabulate" should be placed before "import progressbar"
```

but with a strange twist: it only happens when I do not have my virtualenv activated and call pylint directly: 

```
$ ./.venv/bin/pip install isort==4.3.11
Collecting isort==4.3.11
  Using cached https://files.pythonhosted.org/packages/66/c0/d3c3f902edf8f5f956f26ca0570e59dabfcaee6b75bdd7bdd5a1e8f8f7c9/isort-4.3.11-py2.py3-none-any.whl
Installing collected packages: isort
  Found existing installation: isort 4.3.10
    Uninstalling isort-4.3.10:
      Successfully uninstalled isort-4.3.10
Successfully installed isort-4.3.11

$ ./.venv/bin/pylint cli
************* Module cli.partial_environment_command
cli/partial_environment_command.py:18 [wrong-import-order] third party import "import requests" should be placed before "import botocore.exceptions"
cli/partial_environment_command.py:19 [wrong-import-order] third party import "import tabulate" should be placed before "import botocore.exceptions"
cli/partial_environment_command.py:20 [wrong-import-order] third party import "import troposphere" should be placed before "import botocore.exceptions"
************* Module cli.overview_command
cli/overview_command.py:11 [wrong-import-order] third party import "import tabulate" should be placed before "import progressbar"

-------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 10.00/10, -0.01)


$ ./.venv/bin/pip install isort==4.3.10
Collecting isort==4.3.10
  Using cached https://files.pythonhosted.org/packages/27/ad/b101ff664a1c58589b03291bcaaab2a7df1a1030124f1347fe21b0b94916/isort-4.3.10-py2.py3-none-any.whl
Installing collected packages: isort
  Found existing installation: isort 4.3.11
    Uninstalling isort-4.3.11:
      Successfully uninstalled isort-4.3.11
Successfully installed isort-4.3.10

$ ./.venv/bin/pylint cli

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.99/10, +0.01)

$ source .venv/bin/activate

$ ./.venv/bin/pip install isort==4.3.11
Collecting isort==4.3.11
  Using cached https://files.pythonhosted.org/packages/66/c0/d3c3f902edf8f5f956f26ca0570e59dabfcaee6b75bdd7bdd5a1e8f8f7c9/isort-4.3.11-py2.py3-none-any.whl
Installing collected packages: isort
  Found existing installation: isort 4.3.10
    Uninstalling isort-4.3.10:
      Successfully uninstalled isort-4.3.10
Successfully installed isort-4.3.11

$ ./.venv/bin/pylint cli

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)


$ ./.venv/bin/pip install isort==4.3.10
Collecting isort==4.3.10
  Using cached https://files.pythonhosted.org/packages/27/ad/b101ff664a1c58589b03291bcaaab2a7df1a1030124f1347fe21b0b94916/isort-4.3.10-py2.py3-none-any.whl
Installing collected packages: isort
  Found existing installation: isort 4.3.11
    Uninstalling isort-4.3.11:
      Successfully uninstalled isort-4.3.11
Successfully installed isort-4.3.10

$ ./.venv/bin/pylint cli

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

```

**Comment by Embi** [comment]
I think it is related to #881

**Comment by timothycrosley** [comment]
This should be fixed in 4.3.12: https://github.com/timothycrosley/isort/releases/tag/4.3.12

Thanks!

~Timothy

**Comment by sambrightman** [comment]
Works for me!
